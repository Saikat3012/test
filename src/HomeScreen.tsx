/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React from 'react';
import {
    Text,
    View,
    FlatList,
    StyleSheet,
    TouchableOpacity,
    TextInput,
    Button
} from 'react-native';

import Axios from 'axios';


export default class HomeScreen extends React.Component {
    constructor(props: any) {
        super(props);
        this.state = {
            dataList: [],
            searchDataList: [],
            allPageFetched: false,
            pageNumber: 0,
            searchKeyword: "",
            isSearched: false,
            api: 'https://api.nasa.gov/neo/rest/v1/neo/browse?api_key=Yptb9ngnvn28GF5ZhWuigT2CbVaGGuIg9m4ARnSJ',
        };
    };

    fetchDetails = async () => {
        try {
            const data = await Axios.get(this.state.api);
            this.setState({ dataList: this.state.dataList.concat(data.data.near_earth_objects) })
            console.log(data.data.near_earth_objects)
        } catch (e) {
            console.log(e);
        }

        this.setState({ pageNumber: (this.state.pageNumber) + 1 });
    }
    componentDidMount() {
        this.fetchDetails()
    }
    handleSearch(random: any) {
        if (random) {
            var randomIndex = Math.floor((Math.random() * this.state.dataList.length));
            this.props.navigation.navigate("DetailScreen", this.state.dataList[randomIndex].id)
        } else {
            this.props.navigation.navigate("DetailScreen", this.state.searchKeyword)
        }
    }

    render() {
        const Item = ({ eachData }) => (
            <View style={styles.cardContainer}>
                <Text style={[styles.text, {
                    fontSize: 24,
                }]}>{eachData.name}</Text>
                <Text style={styles.text}>ID : {eachData.id}</Text>
                <Text style={styles.text}>{eachData.nasa_jpl_url}</Text>
                <Text style={styles.text}>{eachData.absolute_magnitude_h}</Text>
            </View>
        );
        const renderItem = ({ item }) => (
            <Item eachData={item} />
        );
        return (
            <View style={styles.container}>
                <View style={{
                    flexDirection: 'row',
                    justifyContent: 'center',
                    alignItems: 'center',
                    width: "100%",
                }}>
                    <View style={styles.searchBarContainer}>

                        <TextInput
                            testId="HomeScreenSearchBar"
                            value={this.state.searchKeyword}
                            placeholder={"Search"}
                            onChangeText={(text) => {
                                this.setState({ searchKeyword: text });
                            }}
                        />

                    </View>
                    <TouchableOpacity
                        testId="HomeScreenSearchButton"
                        style={[{
                            width: "20%",
                            borderRadius: 15,
                            padding: 2,
                            marginLeft: "2%",
                            backgroundColor: "#859",
                        }]}
                        onPress={() => { this.handleSearch(false) }}
                        disabled={this.state.searchKeyword.length == 0 || this.state.dataList.length == 0}>
                        <Text style={styles.text}>Search</Text>
                    </TouchableOpacity>
                </View>
                <TouchableOpacity
                    disabled={this.state.dataList.length == 0}
                    onPress={() => { this.handleSearch(true) }}
                    style={[styles.button]}>
                    <Text style={styles.text}>Random asteroi</Text>
                </TouchableOpacity>
                <FlatList
                    testId="flatList"
                    data={this.state.dataList}
                    renderItem={renderItem}
                    keyExtractor={item => item.id}
                />
            </View>

        )
    }
}

const styles = StyleSheet.create({
    container: {
        paddingHorizontal: 15,
        flex: 1,
        paddingTop: 5,
        alignItems: 'center'
    },
    text: {
        textAlign: 'center',
        color: '#fff',
        fontSize: 18
    },
    searchBarContainer: {
        backgroundColor: '#89E089',
        width: "75%",
        alignSelf: 'center',
        borderRadius: 15,
        paddingHorizontal: 10,

    },
    cardContainer: {
        backgroundColor: '#5FECEC',
        marginVertical: 10,
        padding: 5,
        borderRadius: 15,
        alignItems: 'center',
        justifyContent: 'center',
        width: "100%"
    },
    button: {
        width: "85%",
        justifyContent: 'center',
        alignSelf: 'center',
        marginTop: 7,
        borderRadius: 15,
        backgroundColor: "#859",
        marginBottom:15
    }
})