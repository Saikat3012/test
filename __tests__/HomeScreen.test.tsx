/**
 * @format
 */

import 'react-native';
import React from 'react';
import HomeScreen from '../src/HomeScreen';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
// import Adapter from 'enzyme-adapter-react-16';
import { mount, shallow, configure } from "enzyme";
import Axios from 'axios'
// configure({ adapter: new Adapter() });
const navigation = {
    navigate: jest.fn()
}
test('renders correctly', () => {
    const tree = renderer.create(<HomeScreen />);
    expect(tree).toMatchSnapshot();
});
test('Navigate correctly', () => {
    const tree = renderer.create(<HomeScreen navigation={navigation} />);
    const button = tree.root.findByProps({ testId: "HomeScreenSearchButton" }).props;
    button.onPress();
    expect(navigation.navigate).toBeCalledWith("DetailScreen", "")
});
test('States working correctly', () => {
    const tree = renderer.create(<HomeScreen navigation={navigation} />);
    const textInput = tree.root.findByProps({ testId: "HomeScreenSearchBar" }).props;
    renderer.act(() => textInput.onChangeText("Test"))
    const textInput1 = tree.root.findByProps({ testId: "HomeScreenSearchBar" }).props;
    expect(textInput1.value).toBe("Test")
});
