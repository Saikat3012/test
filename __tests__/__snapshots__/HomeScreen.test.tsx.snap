// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "paddingHorizontal": 5,
      "paddingTop": 5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#89E089",
          "borderRadius": 15,
          "paddingHorizontal": 10,
          "width": "75%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Search"
        rejectResponderTermination={true}
        testId="HomeScreenSearchBar"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 15,
          "marginLeft": "2%",
          "opacity": 1,
          "padding": 2,
          "width": "20%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 22,
            "textAlign": "center",
          }
        }
      >
        Search
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "borderRadius": 15,
        "justifyContent": "center",
        "marginTop": 7,
        "opacity": 1,
        "width": "85%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 22,
          "textAlign": "center",
        }
      }
    >
      Random asteroi
    </Text>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testId="flatList"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
